authors/apps/profiles/views.py

```diff
@@ -67,3 +70,125 @@ class AuthorsAPIView(ListView):
+        return profile
+
+    def delete(self, request, username):
+        """
+        User unfollows an Author
+        """
+        user_id = User.objects.get(username=request.user.username).id
+        try:
+            followed_id = User.objects.get(username=username).id
+            profile_id = Profile.objects.get(user_id=followed_id).id
+            follow = Follow.objects.filter(
+                follower_id=user_id, followed_id=profile_id)
+            if len(follow) == 0:
+                raise Exception(
+                    'You cannot unfollow a user you are not following.')
+            follow.delete()
+            return Response(
+                {'message': 'You have successfully unfollowed {}.'.format(
+                    username)},
+                status=status.HTTP_204_NO_CONTENT
+            )
+        except Exception as e:
+            if isinstance(e, User.DoesNotExist):
+                return Response(
+                    {'error': 'No user with name {} exists.'.format(username)},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+            return Response(
+                {'error': str(e)}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+    def get(self, request, username):
+        """
+        Get all Authors a user is following
+        """
+        user_id = FollowersView().check_user(username)
+        follows = Follow.objects.filter(follower_id=user_id)
+        serializer = FollowSerializer(follows, many=True)
+        following = list()
+        for follow in serializer.data:
+            user_id = Profile.objects.get(id=follow['followed']).user_id
+            username = User.objects.get(id=user_id).username
+            following.append(username)
+        if len(following) == 0:
+            response = {'message': '{} has no following'.format(username)}
+        response = {'following_count': len(following), 'following': following}
+        return Response(response, status=status.HTTP_200_OK)
+
+
+class FollowersView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request, username):
+        """
+        Get all Users following an Author
+        """
+        user_id = self.check_user(username)
+        profile_id = Profile.objects.get(user_id=user_id).id
+        followers = Follow.objects.filter(followed_id=profile_id)
+        serializer = FollowSerializer(followers, many=True)
+        followers = list()
+        for follow in serializer.data:
+            username = User.objects.get(id=follow['follower']).username
+            followers.append(username)
+        if len(followers) == 0:
+            response = {
+                'message': '{} has no followers'.format(username)}
+        else:
+            response = {
+                'follower_count': len(followers), 'followers': followers}
+        return Response(response, status=status.HTTP_200_OK)
+
+    def check_user(self, username):
```

Comment:
Thanks for the good work @joelethan, However, the function naming here does not correspond with what you do in the function. You could consider renaming it to be definitive and relating the name to the function does.